enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx

```diff
@@ -0,0 +1,236 @@
+    id: "TEST_JWT_TOKEN",
+    exp: 1965805007,
+    iat: 1965805007,
+  });
+
+  setupCurrentUserEndpoint(
+    TEST_USER,
+    isValidAuthentication
+      ? undefined
+      : {
+          response: 500,
+        },
+  );
+
+  const settingValues = createMockSettings();
+  const tokenFeatures = createMockTokenFeatures();
+
+  const settingValuesWithToken = {
+    ...settingValues,
+    "token-features": tokenFeatures,
+  };
+  const state = createMockState({
+    settings: mockSettings(settingValuesWithToken),
+    currentUser: TEST_USER,
+  });
+
+  setupEnterprisePlugins();
+  setupApiKeyEndpoints([]);
+  setupSettingsEndpoints([]);
+  setupPropertiesEndpoints(settingValuesWithToken);
+
+  renderWithProviders(<TestComponent authType={authType} {...configOpts} />, {
+    storeInitialState: state,
+    customReducers: SDK_REDUCERS,
+  });
+};
+
+describe("useInitData hook", () => {
+  describe("before authentication", () => {
+    it("should have an error if the authType is incorrect", async () => {
+      setup({ authType: "invalid" });
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "error",
+      );
+    });
+
+    it("should have an error if an API key is not provided", async () => {
+      setup({ authType: "apiKey", apiKey: undefined });
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "error",
+      );
+    });
+
+    it("should have an error if JWT URI is not provided", async () => {
+      setup({ authType: "jwt", jwtProviderUri: undefined });
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "error",
+      );
+    });
+  });
+
+  describe("API Key authentication", () => {
+    it("start loading data if API key and auth type are valid", async () => {
+      setup({ authType: "apiKey" });
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "loading",
+      );
+    });
+
+    it("should set isLoggedIn to true if login is successful", async () => {
+      setup({ authType: "apiKey" });
+      expect(await screen.findByText("Test Component")).toBeInTheDocument();
+
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-is-logged-in",
+        "true",
+      );
+
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "success",
+      );
+    });
+
+    it("should provide an error if login is unsuccessful", async () => {
+      setup({ authType: "apiKey", isValidAuthentication: false });
+
+      expect(await screen.findByTestId("test-component")).toHaveAttribute(
+        "data-login-status",
+        "error",
+      );
+
+      expect(screen.getByTestId("test-component")).toHaveAttribute(
+        "data-error-message",
+        "Could not authenticate: invalid API key",
+      );
```

Comment:
Could we test the error message like this for `before authentication`? Those 3 tests only test `data-login-status="error"`, but never assert the error message. I think it would make testing more clear how they differ from each other.